### PR TITLE
CORS-3417: capi/aws: support ignition proxy and CA bundle

### PR DESCRIPTION
**Pull request by r4f4**
Depends on https://github.com/openshift/installer/pull/8169

/hold

### CODE REVIEW

**Comment by r4f4** [comment]
Update: rebased on top of master.

Waiting on https://github.com/openshift/installer/pull/8142 to run the proxy CI job.

**Comment by r4f4** [comment]
`altinfra-e2e-aws-ovn-proxy`: Install OK and only one e2e failure https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_installer/8170/pull-ci-openshift-installer-master-altinfra-e2e-aws-ovn-proxy/1768933676835409920

/hold cancel

**Comment by r4f4** [comment]
Update: fixed merge conflicts.

**Comment by patrickdillon** [comment]
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/4750/files introduced support for proxies in CAPA, so we can specify the Proxy and CA directly in CAPA rather than ignition

File: pkg/asset/machines/aws/awsmachines.go [hunk 1]
```diff
@@ -160,3 +165,62 @@ func CapaTagsFromUserTags(clusterID string, usertags map[string]string) (capa.Ta
...
+
+	var carefs []capa.IgnitionCASource
+	for len(userCA) > 0 {
+		var block *pem.Block
+		block, userCA = pem.Decode(userCA)
+		if block == nil {
+			return nil, fmt.Errorf("unable to parse certificate, please check the certificates")
+		}
+
+		carefs = append(carefs, capa.IgnitionCASource(dataurl.EncodeBytes(pem.EncodeToMemory(block))))
```
Comment (patrickdillon) [inline pkg/asset/machines/aws/awsmachines.go:199]: Hm, I suspect we don't want `dataurl.EncodeBytes` here, but I do see the e2e proxy test successfully installed.  Do you have any more background on this (no worries if you don't, just checking if I'm missing something)? I would think `dataurl` would be used when passing in ignition, but I wouldn't think that would be needed for the CAs...

Comment (r4f4) [reply pkg/asset/machines/aws/awsmachines.go:199]: As discussed on Slack, it's required as per spec: https://coreos.github.io/ignition/configuration-v3_2/

**Pull request merged**